verifier/src/lib.rs

```diff
@@ -1,17 +1,18 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 
-use air::{ProcessorAir, PublicInputs};
+use air::ProcessorAir;
+pub use air::PublicInputs;
 use core::fmt;
-use vm_core::{ProgramInfo, StackInputs};
-use winterfell::VerifierError;
+pub use vm_core::{Kernel, ProgramInfo, StackInputs};
+pub use winterfell::VerifierError;
```

Comment:
Looks great! Thank you! A couple of questions:

Why do we need to expose `PublicInputs` and `VerifierError`? For `VerifierError` I'm guessing this is for matching against variants of `VerificationError`, but not sure where `PublicInputs` would come in handy.

Also, should we expose a `utils` module from here which would re-export `Serializable`, `Deserializable`, `SliceReader` etc.?

Reply:
Exposing `PublicInputs` was a mistake.  Good catch, thanks!

Yes, the reasoning behind exposing `VerifierError` is exactly as you suggest.

I like your idea of doing a `utils` module.

Btw, here is my silly verifier module I use to check whether I have exposed everything I need to compile:

```rust
use miden_verifier::{verify, Digest, Kernel, ProgramInfo, StackInputs, StackOutputs, StarkProof};
use miden_verifier::math::{Felt};

fn main() {
    let digest0: Digest = Digest::from([Felt::from(0 as u64), Felt::from(0 as u64), Felt::from(0 as u64), Felt::from(0 as u64)]);
    let kernel: Kernel = Kernel::new(&[digest0]);
    let digest: Digest = Digest::from([Felt::from(0 as u64), Felt::from(0 as u64), Felt::from(0 as u64), Felt::from(0 as u64)]);
    let program_info = ProgramInfo::new(digest, kernel);
    let stack_inputs: StackInputs = StackInputs::new(vec![]);
    let stack_outputs: StackOutputs = StackOutputs::new(vec![], vec![]);
    let proof = StarkProof::from_bytes(&[]).unwrap();
    verify(program_info, stack_inputs, stack_outputs, proof).unwrap();
}
```

I didn't need `Serializable`, `Deserializable`, `SliceReader` etc for that example, that's why I didn't expose them.  But exposing them might help with ergonomics?